frontend/src/types/QuestionTypes.ts

```diff
@@ -1,10 +1,41 @@
+import {
+  QuestionTextMetadata,
+  ShortAnswerMetadata,
+  TextMetadata,
+} from "./QuestionMetadataTypes";
+
+export interface QuestionElement {
+  id: string;
+  type: QuestionElementType;
+  data: QuestionElementDataType;
+  error?: string;
+}
+
 export enum QuestionElementType {
-  QUESTION = "Question",
-  TEXT = "Text",
-  IMAGE = "Image",
-  MULTIPLE_CHOICE = "Multiple Choice",
-  SHORT_ANSWER = "Short Answer",
-  MULTI_SELECT = "Multi-select",
+  QUESTION_TEXT = "QUESTION_TEXT",
+  TEXT = "TEXT",
+  IMAGE = "IMAGE",
+  MULTIPLE_CHOICE = "MULTIPLE_CHOICE",
+  SHORT_ANSWER = "SHORT_ANSWER",
+  MULTI_SELECT = "MULTI_SELECT",
+}
+
+export type QuestionElementDataType =
+  // remove string when all the elements have been implemented
```

Comment:
There's some placeholder values for other elements right now, so leaving this in. Later, when we have all the elements implemented, can remove this line!